From 96207cac60ddf5e77d2cca2022d89ae7857b1fc7 Mon Sep 17 00:00:00 2001
From: TungDang <tung.dang.jy@rvc.renesas.com>
Date: Sun, 21 Aug 2016 07:20:23 +0700
Subject: [PATCH 4/9]
 gst145-waylandsink-Restrict-the-number-of-buffers-in-the-pool to be allocated

This patch sets the maximum number of buffers to be allocated in
the buffer pool waylandsink has.

Porting from a patch in Linux 3.10 environment made by Kazunori Kobayashi <kkobayas@igel.co.jp>

Signed-off-by: TungDang <tung.dang.jy@rvc.renesas.com>
---
 ext/wayland/gstwaylandsink.c | 6 +++---
 ext/wayland/waylandpool.h    | 1 +
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/ext/wayland/gstwaylandsink.c b/ext/wayland/gstwaylandsink.c
index a40b7e0..211bb6a 100644
--- a/ext/wayland/gstwaylandsink.c
+++ b/ext/wayland/gstwaylandsink.c
@@ -592,7 +592,7 @@ gst_wayland_sink_set_caps (GstBaseSink * bsink, GstCaps * caps)
     goto pool_failed;

   structure = gst_buffer_pool_get_config (newpool);
-  gst_buffer_pool_config_set_params (structure, caps, info.size, 2, 0);
+  gst_buffer_pool_config_set_params (structure, caps, info.size, GST_WAYLAND_BUFFER_POOL_NUM, GST_WAYLAND_BUFFER_POOL_NUM);
 #ifdef HAVE_WAYLAND_KMS
   gst_structure_set (structure, "videosink_buffer_creation_request_supported",
       G_TYPE_BOOLEAN, TRUE, NULL);
@@ -686,7 +686,7 @@ gst_wayland_sink_propose_allocation (GstBaseSink * bsink, GstQuery * query)
     size = info.size;

     config = gst_buffer_pool_get_config (pool);
-    gst_buffer_pool_config_set_params (config, caps, size, 2, 0);
+    gst_buffer_pool_config_set_params (config, caps, size, GST_WAYLAND_BUFFER_POOL_NUM, GST_WAYLAND_BUFFER_POOL_NUM);
 #ifdef HAVE_WAYLAND_KMS
     gst_structure_set (config, "videosink_buffer_creation_request_supported",
         G_TYPE_BOOLEAN, TRUE, NULL);
@@ -696,7 +696,7 @@ gst_wayland_sink_propose_allocation (GstBaseSink * bsink, GstQuery * query)
       goto config_failed;
   }
   if (pool) {
-    gst_query_add_allocation_pool (query, pool, size, 2, 0);
+    gst_query_add_allocation_pool (query, pool, size, GST_WAYLAND_BUFFER_POOL_NUM, GST_WAYLAND_BUFFER_POOL_NUM);
     /*
      * Add the default allocator for the plugins that can't use dmabuf
      * descriptors.
diff --git a/ext/wayland/waylandpool.h b/ext/wayland/waylandpool.h
index e0944d0..3523d24 100644
--- a/ext/wayland/waylandpool.h
+++ b/ext/wayland/waylandpool.h
@@ -32,6 +32,7 @@ G_BEGIN_DECLS
 #define GST_IS_WAYLAND_BUFFER_POOL(obj)   (G_TYPE_CHECK_INSTANCE_TYPE ((obj), GST_TYPE_WAYLAND_BUFFER_POOL))
 #define GST_WAYLAND_BUFFER_POOL(obj)      (G_TYPE_CHECK_INSTANCE_CAST ((obj), GST_TYPE_WAYLAND_BUFFER_POOL, GstWaylandBufferPool))
 #define GST_WAYLAND_BUFFER_POOL_CAST(obj) ((GstWaylandBufferPool*)(obj))
+#define GST_WAYLAND_BUFFER_POOL_NUM 3

 typedef struct _GstWaylandBufferPool GstWaylandBufferPool;
 typedef struct _GstWaylandBufferPoolClass GstWaylandBufferPoolClass;
--
1.9.1
